src/main/java/CLI/Main.java

```diff
@@ -375,4 +387,91 @@ private static void checkHistoryAddJobsAndDownload(Job job, boolean removeInputB
             downloader.run();
         }
     }
-}
+
+    public static boolean checkUpdate() throws URISyntaxException {
+        String latestVersion = getLatestVersion();
+        if (isNewerVersion(latestVersion , VERSION_NUMBER)) {
+            String Link;
+            String oldFilePath = String.valueOf(Main.class.getProtectionDomain().getCodeSource().getLocation().toURI());
+            String newFilePath = "";
+            String OS_NAME = OS.getOSName();
+            if (OS_NAME.contains("win")) {
+                Link = "https://github.com/SaptarshiSarkar12/Drifty/releases/latest/download/Drifty-CLI.exe";
+                String fileName = "Drifty_CLI.exe";
+                String dirPath = String.valueOf(Paths.get(System.getenv("LOCALAPPDATA"), "Drifty", "updates"));
+                FileDownloader downloader =  new FileDownloader(Link , fileName , dirPath);
+                downloader.run();
+                newFilePath = dirPath +'\\' + fileName;
+            } else if (OS_NAME.contains("mac")) {
+                Link = "https://github.com/SaptarshiSarkar12/Drifty/releases/latest/download/Drifty-CLI_macos";
+                String fileName = "Drifty-CLI_macos";
+                String dirPath = ".drifty/updates";
+                FileDownloader downloader =  new FileDownloader(Link , fileName , dirPath);
+                downloader.run();
+                newFilePath = dirPath +'\\' + fileName;
+            } else if (OS_NAME.contains("linux")) {
+                Link = "https://github.com/SaptarshiSarkar12/Drifty/releases/latest/download/Drifty-CLI_linux";
+                String fileName = "Drifty-CLI_linux";
+                String dirPath = ".drifty/updates";
+                FileDownloader downloader = new FileDownloader(Link, fileName, dirPath);
+                downloader.run();
+                newFilePath = dirPath +'\\' + fileName;
+            }
+            Updater.replaceUpdate(oldFilePath , newFilePath);
+            return true;
+        }
+        return false;
+    }
+
+    private static String getLatestVersion() {
+        try {
+            URL url = new URI("https://api.github.com/repos/SaptarshiSarkar12/Drifty/releases/latest").toURL();
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            connection.setRequestMethod("GET");
+
+            BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+            String response = reader.readLine();
+            reader.close();
+
+            // Parse JSON response to get the "tag_name"
+            // For simplicity, we're assuming the tag_name is a string enclosed in double quotes
+            int start = response.indexOf("\"tag_name\":\"") + 12;
```

Comment:
@kmr-rohit Try using GSon to get the value of the exact element in the JSON output. Do not use indent because if GitHub api results change then it would become hectic to count the indent and indices for the tag element.
You can check [this stackoverflow answer](https://stackoverflow.com/a/23359047) to know how you can do the same.

Reply:
Replaced fetching response value using JsonParser rather than exact Json Output.

Reply:
Okay :+1: